### DISCUSSION TITLE
Datumaro with only annotations?

### DISCUSSION THREAD

**Discussion by knelson-hme** [openvinotoolkit/datumaro#1231] [Q&A]
Is it possible to use Datumaro with annotations only and no associated image files? I have a fairly large video file I am annotating in CVAT which I then export into Datumaro format without the "Save images" checked in CVAT (I do this because the resulting file exported from CVAT becomes very large if I do). I then want to do some post processing on the annotations, but I run into some issues when I try and run the following simple code:

```
data_path = "path/to/dataset"
dataset = Dataset.import_from(data_path)
for item in dataset:
    print(item)
```

It seems like the `import_from` method is working but when I try to loop through the dataset I get issues as Datumaro is trying to find an associated image file even though one doesn't exist. When I repeat the same steps above with "Save images" check in CVAT everything works, but this takes a lot of extra time and space to do as it creates a jpg of all my frames.

I have been able to loop through my annotations using the following dirty way, but I am sure there has to be a better way.

```
data_path = "path/to/dataset"
dataset = Dataset.import_from(data_path)
for item in dataset._data._source:
    print(item)
```

**Comment by vinnamkim** [comment]
Hi @knelson-hme,
I guess your problem happens here:
https://github.com/openvinotoolkit/datumaro/blob/cce5fc9ea1ccfa3954f785c338ca1fe823f089a2/src/datumaro/plugins/data_formats/cvat/base.py#L367-L376
There is a file existence check during importing CVAT data format. It requires there exist actual image files in `images_dir` which is
https://github.com/openvinotoolkit/datumaro/blob/cce5fc9ea1ccfa3954f785c338ca1fe823f089a2/src/datumaro/plugins/data_formats/cvat/base.py#L60-L63

To resolve this, you might create a soft directory link or copy-and-paste them to put your images in the right place (`images_dir`). 

You can refer to the directory structure of CVAT data format here: https://openvinotoolkit.github.io/datumaro/latest/docs/data-formats/formats/cvat.html#import-cvat-dataset
![image](https://github.com/openvinotoolkit/datumaro/assets/26541465/b5748331-eda1-431c-9d6a-778361a0c3e9)

Unfortunately, there is currently no functionality to assign `images_dir` outside. We will put this functionality on our backlog.

Thanks!

**Comment by knelson-hme** [comment]
Thank you for taking the time to read and reply to my post.

I just want to be clear that my dataset is in Datumaro format and not CVAT when I export it from the CVAT GUI interface.

When I run my code I get a problem with the following lines of code:

https://github.com/openvinotoolkit/datumaro/blob/cce5fc9ea1ccfa3954f785c338ca1fe823f089a2/src/datumaro/components/dataset_storage.py#L259-L264

Where it hits the exception and says 
```
datumaro.components.errors.MediaTypeError: Unexpected media type of a dataset item 'frame_000000'. Expected '<class 'datumaro.components.media.Image'>', actual '<class 'datumaro.components.media.PointCloudFromFile'>'
```

From my understanding this error is happening because I don't have any frame image data saved with my annotations. From your reply it seems like Datumaro requires image frame data to be located with the annotations.

Let me explain my original intention to hopefully clear up any confusion. I want to annotate a series of videos that are about 1 hour long each. I import the video into CVAT's GUI and do my annotations. In order to save space I was hoping I could just export the annotations in Datumaro format without also generating a bunch of individual jpeg frames which take up a lot of space. This could be the stupid step, but I then want to use my own python code to load the original video using opencv and also read in the Datumaro dataset annotations (that don't have any explicit image data saved with it) and do some post processing on annotations. My thinking was that I could use opencv to generate the individual frames from the original video file and that these should line up with the Datumaro annotations. I was thinking I could do it this way so I only had to have one copy of the original video without any extra jpeg frames saved to disk. Maybe this approach is flawed and I shouldn't be trying to do it this way. I am curious to know your thoughts.

**Reply by vinnamkim** [reply]
Hi You're welcome,

This seems a bug since I can see that the media type is `PointCloud` from your log,
`datumaro.components.errors.MediaTypeError: Unexpected media type of a dataset item 'frame_000000'. Expected '<class 'datumaro.components.media.Image'>', actual '<class 'datumaro.components.media.PointCloudFromFile'>'`.

Sorry for inconvenience, but long ago, CVAT forked out this repository and has been maintaining [its own forked Datumaro](https://github.com/cvat-ai/datumaro). We might need to contact the CVAT team and recheck the video dataset exportation workflow at the same time.

>  Maybe this approach is flawed and I shouldn't be trying to do it this way. I am curious to know your thoughts.

I don't know what is your next workflow for the exported dataset exactly. However, for your question, my thought is that you would better to export your dataset with jpeg frames, if your next workflow requires multiple accesses for the video frames (e.g., model training). This is because you know video decoding costs enormously expensive computation.

**Reply by vinnamkim** [reply]
Root cause of this problem: https://github.com/opencv/cvat/issues/5924

**Comment by ianbenlolo** [comment]
This would be super useful. Just being able to convert from one type to another without images (ex coco<-->cvat, etc) without having to download/format the images in a specific structure.
